x-pack/plugins/reporting/public/components/shared/get_shared_components.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CoreSetup } from 'kibana/public';
+import React from 'react';
+import { ReportingAPIClient } from '../..';
+import { PDF_REPORT_TYPE } from '../../../common/constants';
+import type { Props as PanelPropsScreenCapture } from '../screen_capture_panel_content';
+import { ScreenCapturePanelContent } from '../screen_capture_panel_content_lazy';
+
+interface IncludeOnCloseFn {
+  onClose: () => void;
+}
+
+type PropsPDF = Pick<PanelPropsScreenCapture, 'getJobParams' | 'layoutOption'> & IncludeOnCloseFn;
+
+export function getSharedComponents(core: CoreSetup) {
+  return {
+    ReportingPanelPDF(props: PropsPDF) {
+      return (
+        <ScreenCapturePanelContent
+          layoutOption={props.layoutOption}
+          requiresSavedState={false}
```

Comment:
I agree with you it's cleaner to treat this as if it could be used in Dashboard. I don't want to get too deep yet in perfecting the integration touchpoints for the reporting panel. There will be future tech debt / refactoring work to move the export implementation logic out of Reporting, and have new plugins that provide export implementation. That will affect the integration touchpoints, because apps would go from importing this panel component from Reporting, to importing it from a new plugin that controls the implementation detail of the export.

- current: `<services.reporting.components.ReportingPanelPDF />`
- future: `<services.exportPDF.components.ReportingPanelPDF />`

That is why this PR does not change the Reporting plugin contract further, for example it doesn't add CSV or PNG export panel to the contract. 

This is a good note for any reviewer - I'll update the description to talk about this.

cc @ppisljar

Reply:
> There will be future tech debt / refactoring work to move the export implementation logic out of Reporting, and have new plugins that provide export implementation

Related Discuss issue: https://github.com/elastic/kibana/issues/101422